crates/flux-refineck/src/checker.rs

```diff
@@ -1543,7 +1578,7 @@ impl<'ck, 'genv, 'tcx, M: Mode> Checker<'ck, 'genv, 'tcx, M> {
             && let (deref_ty, alloc_ty) = args.box_args()
             && let TyKind::Indexed(BaseTy::Array(arr_ty, arr_len), _) = deref_ty.kind()
         {
-            let idx = Expr::from_const(self.genv.tcx(), arr_len);
+            let idx = Expr::from_const(self.genv.tcx(), &arr_len);
```

Comment:
```suggestion
            let idx = Expr::from_const(self.genv.tcx(), arr_len);
```

Reply:
I swear everytime I add the `&` rustc says to remove it, and then when I remove it clippy says to add it back (or the other way around!!!)

Reply:
That's probably rust-analyzer, you have to disable native diagnostic